src/Illuminate/Console/Scheduling/Event.php

```diff
@@ -597,15 +605,42 @@ public function pingOnFailure($url)
      */
     protected function pingCallback($url)
     {
-        return function (Container $container, HttpClient $http) use ($url) {
+        return function (Container $container) use ($url) {
             try {
-                $http->request('GET', $url);
+                $this->getHttpClient($container)->request('GET', $url);
             } catch (ClientExceptionInterface|TransferException $e) {
                 $container->make(ExceptionHandler::class)->report($e);
             }
         };
     }
 
+    /**
+     * Get the Guzzle HTTP Client to use to send pings.
+     *
+     * @param  \Illuminate\Contracts\Container\Container  $container
+     * @return \GuzzleHttp\ClientInterface
+     */
+    protected function getHttpClient(Container $container): HttpClientInterface
+    {
+        if (null !== $this->httpClient) {
+            return $this->httpClient;
+        }
+
+        if ($container->bound(HttpClientInterface::class)) {
+            return $this->httpClient = $container->make(HttpClientInterface::class);
+        }
+
+        if ($container->bound(HttpClient::class)) {
```

Comment:
This is necessary so that instead of just calling make and getting `new HttpClient`, we can construct it ourselves, below. We only wanna go get from the container if it is actually bound there.